suzieq/poller/worker/nodes/node.py

```diff
@@ -1187,6 +1188,93 @@ async def rest_gather(self, service_callback, cmd_list, cb_token,
 class IosXENode(Node):
     '''IOS-XE Node-sepcific telemetry gather implementation'''
 
+    async def _init_ssh(self, init_boot_time=True, rel_lock: bool = True) -> None:
+        '''Need to start an interactive session for XE
+
+        Many IOSXE devices cannot accept commands as fast as we can fire them.
+        Furthermore, many TACACS servers also collapse under the load of
+        multiple authemtication requests because each SSH session needs to be
+        authenticated, not just the main SSH connection.
+        '''
+        backoff_period = 1
+        self.WAITFOR = r'.*[>#]\s*$'
+
+        while not self._conn:
+            await super()._init_ssh(init_boot_time=False, rel_lock=False)
+            if not self._conn:
+                await asyncio.sleep(backoff_period)
+                backoff_period *= 2
+                backoff_period = min(backoff_period, 120)
+                if rel_lock:
+                    self.ssh_ready.release()
+                return
+
+        if self._conn:
+            try:
+                self._stdin, self._stdout, self._stderr = \
+                    await self._conn.open_session(term_type='xterm')
+                self.logger.info(
+                    f'Persistent SSH created for {self.hostname}')
+
+                output = await self.wait_for_prompt()
+                if output.strip().endswith('>'):
+                    # We're not in enable mode and cannot function
+                    self.logger.warning(
+                        'Privilege escalation required. Trying')
```

Comment:
Add self.hostname in logs for better undestanding

Reply:
Done